### PR TITLE
(feat): OpenEBS pool disk loss chart

### PR DESCRIPTION
**Pull request by imrajdas**
- Adding openEBS pool disk loss chart
- Issue- https://github.com/litmuschaos/litmus/issues/1193

### CODE REVIEW

**Comment by imrajdas** [comment]
[pool-disk-loss.txt](https://github.com/litmuschaos/chaos-charts/files/4331498/pool-disk-loss.txt)

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+            # Enter the APP_NAMESPACE, if you set APP_CHECK to true
+            - name: APP_NAMESPACE
+              value: ''
+              
+            - name: APP_PVC
+              value: ''
+
+            # This is a chaos namespace which will create all infra chaos resources in that namespace
+            - name: CHAOS_NAMESPACE
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml:30]: We don't support downward api yet in chaosEngine. cc: @ispeakc0de

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                  fieldPath: metadata.namespace
+
+            # Only GCP and AWS is supported
+            - name: CLOUD_PLATFORM
+              value: 'GCP'
+
+            # Enter the time duration in sec after the disk is dettached
+            - name: TOTAL_CHAOS_DURATION
+              value: '10'
+
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml:40]: Is this the default duration we are using for other exp?

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            # Enter the project id for gcp only
+            - name: PROJECT_ID 
+              value: ''
+
+            # Enter the node name
+            - name: NODE_NAME
+              value: '' 
+
+            # Enter the disk name
+            - name: DISK_NAME
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml:54]: The approach we are taking in sample engines is to provide values. See sample engine for generic disk loss: https://github.com/litmuschaos/chaos-charts/blob/master/charts/generic/disk-loss/engine.yaml

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+          - "litmuschaos.io"
+          - "batch"
+          - "extensions"
+          - "storage.k8s.io"
+          - "openebs.io"
+        resources:
+          - "pods"
+          - "jobs"
+          - "events"
+          - "pods/exec"
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/experiment.yaml:29]: Dont we need pod/log too now?

File: charts/openebs/openebs-pool-disk-loss/openebs-pool-disk-loss.chartserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  version: 0.1.0
+  annotations:
+    categories: OpenEBS
+    vendor: CNCF
+    support: https://slack.openebs.io/
+spec:
+  displayName: openebs-pool-disk-loss 
+  categoryDescription: >
+    OpenEBS Pool Disk Loss contains chaos to disrupt state of infra resources. Experiments can inject disk loss against openEBS pool.
+    - Causes (forced/graceful) disk loss from the node pool.
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/openebs-pool-disk-loss.chartserviceversion.yaml]: You mean node disk pool

File: charts/openebs/openebs-pool-disk-loss/openebs-pool-disk-loss.chartserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Source Code
+      url: https://github.com/litmuschaos/litmus/tree/master/experiments/openebs/openebs-pool-disk-loss
+    - name: Documentation
+      url: https://docs.litmuschaos.io/docs/openebs-pool-disk-loss/
+    - name: Video
+      url:
+  icon:
+    - url: ""
+      mediatype: ""
+  chaosexpcrdlink: https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/openebs/openebs-pool-disk-loss/experiment.yaml
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/openebs-pool-disk-loss.chartserviceversion.yaml]: Are we ising this still nowadays - don't see it on others?

File: charts/openebs/openebs-pool-disk-loss/rbac.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+# Source: openebs/templates/clusterrole.yaml
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: pool-disk-loss-sa
+  labels:
+    name: pool-disk-loss-sa
+rules:
+- apiGroups: ["","apps","litmuschaos.io","batch","extensions","storage.k8s.io","openebs.io"]
+  resources: ["pods", "jobs", "events", "pods/exec", "cstorpools", "configmaps", "secrets", "storageclasses", "persistentvolumes", "persistentvolumeclaims", "cstorvolumereplicas", "chaosexperiments", "chaosresults", "chaosengines"]
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/rbac.yaml]: pod/log

**Review by ksatchit — CHANGES_REQUESTED** [review]
Shared comments!

File: charts/openebs/openebs-pool-disk-loss/rbac.yaml (hunk 7 above)
Comment (imrajdas) [reply charts/openebs/openebs-pool-disk-loss/rbac.yaml]: It's working without the pod/log

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml (hunk 4 above)
Comment (imrajdas) [reply charts/openebs/openebs-pool-disk-loss/experiment.yaml:29]: It's working without the pod/log

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+spec:
+  # It can be true/false
+  annotationCheck: 'false'
+  # It can be active/stop
+  engineState: 'active'
+  #ex. values: ns1:name=percona,ns2:run=busybox 
+  auxiliaryAppInfo: ''
+  appinfo:
+    appns: 'default'
+    applabel: 'app=busybox'
```
Comment (ispeakc0de) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml]: app=nginx

File: charts/openebs/openebs-pool-disk-loss/engine.yaml (hunk 2 above)
Comment (imrajdas) [reply charts/openebs/openebs-pool-disk-loss/engine.yaml:40]: Change to 15

File: charts/openebs/openebs-pool-disk-loss/engine.yaml (hunk 3 above)
Comment (imrajdas) [reply charts/openebs/openebs-pool-disk-loss/engine.yaml:54]: Done

File: charts/openebs/openebs-pool-disk-loss/openebs-pool-disk-loss.chartserviceversion.yaml (hunk 5 above)
Comment (imrajdas) [reply charts/openebs/openebs-pool-disk-loss/openebs-pool-disk-loss.chartserviceversion.yaml]: Done

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  chaosServiceAccount: pool-disk-loss-sa
+  monitoring: false
+  # It can be delete/retain
+  jobCleanUpPolicy: 'delete'
+  experiments:
+    - name: openebs-pool-disk-loss
+      spec:
+        components:
+          env:
+            - name: ANSIBLE_STDOUT_CALLBACK
```
Comment (ispeakc0de) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml]: Not needed

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+              value: 'default'
+
+            # Enter the App_label, if you set APP_CHECK to true
+            - name: APP_LABEL
+              value: 'app=nginx'
+
+            # Enter the APP_NAMESPACE, if you set APP_CHECK to true
+            - name: APP_NAMESPACE
+              value: 'default'
+
```
Comment (ispeakc0de) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml]: not needed

File: charts/openebs/openebs-pool-disk-loss/engine.yaml (hunk 2 above)
Comment (ispeakc0de) [reply charts/openebs/openebs-pool-disk-loss/engine.yaml:40]: We are using 60s i guess

File: charts/openebs/openebs-pool-disk-loss/engine.yaml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+            - name: DEVICE_NAME
+              value: '/dev/sdb'
+
+            # Enter the zone name
+            - name: ZONE_NAME
+              value: 'us-central1-a' 
+
+            # provide auxiliary application details - namespace and labels of the applications
+            # sample input is - "ns1:app=percona,ns2:name=nginx"
+            - name: AUXILIARY_APPINFO
```
Comment (ispeakc0de) [inline charts/openebs/openebs-pool-disk-loss/engine.yaml]: Not needed

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml [hunk 12]
```diff
@@ -0,0 +1,110 @@
+---
+apiVersion: litmuschaos.io/v1alpha1
+description:
+  message: |
+    Kill OpenEBS pool disk loss.
```
Comment (ispeakc0de) [inline charts/openebs/openebs-pool-disk-loss/experiment.yaml]: Can we add more description

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml (hunk 4 above)
Comment (imrajdas) [reply charts/openebs/openebs-pool-disk-loss/experiment.yaml:29]: Okay Got it why we need pod/log. Added the permission

Comment (ispeakc0de) [reply charts/openebs/openebs-pool-disk-loss/experiment.yaml:29]: Yes we need pods/log, to get the failure step if it fails

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+      - name: ZONE_NAME
+        value: ''
+
+      - name: DEVICE_NAME
+        value: ''
+
+      # provide auxiliary application details - namespace and labels of the applications
+      # sample input is - "ns1:app=percona,ns2:name=nginx"
+      - name: AUXILIARY_APPINFO
```
Comment (ispeakc0de) [inline charts/openebs/openebs-pool-disk-loss/experiment.yaml]: Not needed

**Review by ispeakc0de — CHANGES_REQUESTED** [review]

**Comment by ispeakc0de** [comment]
@rajdas98 Can you add this experiment in openebs package.yaml file also

File: charts/openebs/openebs-pool-disk-loss/engine.yaml (hunk 1 above)
Comment (ispeakc0de) [reply charts/openebs/openebs-pool-disk-loss/engine.yaml:30]: Can you please removed this downward api?

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml (hunk 13 above)
Comment (ispeakc0de) [reply charts/openebs/openebs-pool-disk-loss/experiment.yaml]: Can you please remove this?

**Review by ispeakc0de — CHANGES_REQUESTED** [review]

File: charts/openebs/openebs-pool-disk-loss/experiment.yaml [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+    - -c
+    - ansible-playbook ./experiments/openebs/openebs-pool-disk-loss/openebs_pool_disk_loss_ansible_logic.yml -i /etc/ansible/hosts -vv; exit 0
+    command:
+    - /bin/bash
+    env:
+      - name: ANSIBLE_STDOUT_CALLBACK
+        value: 'default'
+
+      - name: TOTAL_CHAOS_DURATION
+        value: '15'
```
Comment (ksatchit) [inline charts/openebs/openebs-pool-disk-loss/experiment.yaml]: 60?

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]
/lgtm

**Pull request merged**